api/key/key.go

```diff
@@ -0,0 +1,23 @@
+package key
+
+import (
+	"github.com/open-telemetry/opentelemetry-go/api/core"
+	"github.com/open-telemetry/opentelemetry-go/api/registry"
+)
+
+type AnyValue struct{}
+
+func (AnyValue) String() string {
+	return "AnyValue"
+}
+
+func New(name string, opts ...registry.Option) core.Key {
+	return core.Key{
+		Variable: registry.Register(name, AnyValue{}, opts...),
+	}
+}
+
+var (
+	WithDescription = registry.WithDescription
+	WithUnit        = registry.WithUnit
```

Comment:
should we use function instead of global var?
and I found `api/stats/stats.go` and `api/key/key.go` have the same duplicate code.
you said `The core package shrinks to cover only SpanContext, Key, Value, and KeyValue.`, so why add `api/key` package? thanks a lot!

Reply:
> should we use function instead of global var?
> and I found `api/stats/stats.go` and `api/key/key.go` have the same duplicate code.

Yes. (Sorry, I was briefly overcome by laziness.)

> you said `The core package shrinks to cover only SpanContext, Key, Value, and KeyValue.`, so why add `api/key` package? thanks a lot!

I think this is all still up for debate, but I was trying to create a common base type for metrics and keys, both of which are  _sometimes_ registered. The user wouldn't use this common type (`Variable`), but would use the `Key` type directly whenever constructing `KeyValue`. To summarize, `Key` is for making `KeyValue` whereas `Variable` is the common base of a thing that can be registered.

Reply:
got it, thank your explain!